va/va_enc_av1.h

```diff
@@ -0,0 +1,974 @@
+            /** \brief Corresponds to AV1 syntax element of the same name. */
+            uint32_t    enable_restoration                          : 1;
+            /** \brief Reserved bytes for future use, must be zero. */
+            uint32_t    reserved_bits                               : 18;
+        } bits;
+        uint32_t value;
+    } seq_fields;
+
+    /** \brief Corresponds to AV1 syntax element of the same name.
+     *  value range [0..7].
+     */
+    uint8_t     order_hint_bits_minus_1;
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t                va_reserved[VA_PADDING_HIGH];
+} VAEncSequenceParameterBufferAV1;
+
+#define VA_AV1_MAXNUM_SEGMENTS                  8
+#define VA_AV1_MAXNUM_SEGFEATURES               8
+
+/**
+ * \brief Segment parameters
+ */
+typedef struct _VAEncSegParamAV1 {
+    union {
+        struct {
+            /** \brief Indicates if segmentation is enabled in the current frame.
+             *  If disabled, all the below parameters in the structure should
+             *  be set to 0, and ignored by driver.
+             */
+            uint8_t     segmentation_enabled            : 1;
+
+            /** \brief when segmentation_enabled equals 1 and segment_number > 0,
+             *  this parameter equals 1 indicates the segmentation map may
+             *  come from DDI, and that the Segmentation Map Data Buffer
+             *  should be provided with populated segment_id. If equals 0,
+             *  segmentation map should be inherited from a reference frame
+             *  (specified by primary_ref_frame). When segmentation_enabled or
+             *  SegmentNumber equals 0, this parameter should be set to 0
+             *  and ignored by driver.
+             */
+            uint8_t     segmentation_update_map         : 1;
+            uint8_t     segmentation_temporal_update    : 1;
+            /** \brief Reserved bytes for future use, must be zero. */
+            uint8_t     reserved                        : 5;
+
+        } bits;
+        uint8_t value;
+    } seg_flags;
+
+    /** \brief If segmentation_enabled equals 1, this parameter indicates
+     *  the number of segments conveyed through VAAPI. In this case,
+     *  if segment_number equals 0, it forces the underline encoder
+     *  to determine how many segments would be created as well as
+     *  the segmentation map to be generated. Also the underline
+     *  encoder shall write the segmentation_params() syntax in the
+     *  uncompressed header at segmentation_bitoffset (back-annotation).
+     *  In VAAPI, the rest parameters in this structure should be all
+     *  set to 0 and ignored by driver. And app should NOT send the
+     *  "Segment map data buffer". In packed uncompressed header
+     *  bitstream, app should write syntax element segmentation_enabled
+     *  as 0 and segmentation_params() should be only 1-bit-long.
```

Comment:
dont understand whether when the segment_number ==0, encoder will enable it or not. it looks different with previous description.

Reply:
when segmentation_enabled == 1 and segment_number ==0,  driver will enable it with segment params generated by driver rather from application.

Reply:
Add more detailed comments for explanation.